src/ast/FieldDecl.ts

```diff
@@ -12,12 +12,39 @@ export class FieldDecl extends AstNode {
 
     fields: VarList;
     modifier: string;
-    generateGetter: boolean;
-    generateSetter: boolean;
+    generateGetter: boolean = false;
+    generateSetter: boolean = false;
 
     public parse(context: Tokenizer): any {
-        // TODO: implement the rest.
+        let indentLevel: number = context.getCurrentLineTabLevel();
+        context.getAndCheckNext("fields");
+        this.modifier = context.getNext();
         this.fields = new VarList();
+        this.fields.parse(context);
+
+        if (context.getCurrentLineTabLevel() <= indentLevel) {
+            return;
+        }
+
+        context.getAndCheckNext("generate");
+
+        if (context.getNext() === "getters") {
```

Comment:
Probably best to save this into a variable and verify that variable is either "getters" or "setters" and to throw if its neither.

Reply:
Good call! Would just also like to plug that Tokenizer exposes some constants for keywords if anyone wants to use those (e.g. Tokenizer.literals.Getters.val)